src/curves/ed25519Bip32.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  stringToBytes,
+  bytesToHex,
+  hexToBigInt,
+  remove0x,
+} from '@metamask/utils';
+import { mod } from '@noble/curves/abstract/modular';
+import { ed25519 } from '@noble/curves/ed25519';
+
+export const curve = ed25519.CURVE;
+
+/**
+ * Named after whitepaper: BIP32-Ed25519 Hierarchical Deterministic Keys over a Non-linear Keyspace
+ * https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf
+ * "vanilla" "ed25519" curve follows SLIP10: https://tezos.stackexchange.com/questions/2837/can-i-use-bip32-hd-key-pairs-to-derive-ed25519-addresses
+ * note that that the important difference of the "bip32" version is that it allows unhardened key derivation
+ */
+export const name = 'ed25519Bip32';
```

Comment:
Should we call this "bip32-ed25519" (since that's what the spec is called)?

Reply:
We had an internal discussion about this too. 

Regarding the `-`, we kept camel case, this is later imported with `import * as ed25519Bip32` which does not work with kebab case (afaik), and also is then put into object, which allows only camel case (we can disable eslint for that line ofc, but we decided that it's not worth it)

Initially, I called this `bip32Ed25519` but after discussions changed it to `ed25519Bip32` because 
> bip32 is "semantically" less important than the curve itself

Let me know what you think :)